marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -659,68 +663,108 @@ runIndexers
+        getStartingPointsFromIndexers indexerDepth currentNodeBlockNo indexers coordinator
+      let oldestCommonChainPoint = minimum resumablePoints
+          resumePoint = case cliChainPoint of
+            C.ChainPointAtGenesis -> oldestCommonChainPoint -- User didn't specify a chain point, use oldest common chain point,
+            cliCp -> cliCp -- otherwise use what was provided on CLI.
+      logInfo trace $
+        renderStrict $
+          layoutPretty defaultLayoutOptions $
+            "Resumable points for each indexer:"
+              <> line
+              <> indent 4 (align (list (fmap pretty resumablePoints)))
+
+      -- Possible runtime failure if an indexer with a non-genesis resumable point will resume from
+      -- genesis.
+      if shouldFailIfResync
+        && elem C.ChainPointAtGenesis resumablePoints
+        && any (\case ChainPoint{} -> True; _ -> False) resumablePoints
+        then do
+          logError trace $
+            renderStrict $
+              layoutPretty defaultLayoutOptions $
+                nest
+                  4
+                  ( "At least one indexer has a non-genesis resumable point, while the oldest common resumable point between indexers is genesis."
+                      <> line
+                      <> "Are you sure you want to restart syncing that indexer from genesis?"
+                      <> line
+                      <> "If so, remove the '--fail-if-resyncing-from-genesis' flag."
+                  )
+        else do
+          let stream =
+                mkIndexerStream coordinator
+                  . chainSyncEventStreamLogging trace
+                  . updateProcessedBlocksMetric
+              io = withChainSyncEventEpochNoStream socketPath networkId [resumePoint] stream
+              handleException NoIntersectionFound = do
+                logError trace $
+                  renderStrict $
+                    layoutPretty defaultLayoutOptions $
+                      "No intersection found when looking for the chain point"
+                        <+> pretty resumePoint
+                        <> "."
+                        <+> "Please check the slot number and the block hash do belong to the chain."
+                signalQSemN (coordinator ^. barrier) (coordinator ^. indexerCount)
+           in finally (io `catch` handleException) (waitForIndexersToFinishProcessingLastEvent trace coordinator)
+
+    waitForIndexersToFinishProcessingLastEvent :: Trace IO Text -> Coordinator' a -> IO ()
+    waitForIndexersToFinishProcessingLastEvent trace coordinator = do
+      let secondsBeforeTimeout = 180
+      logInfo trace $
+        "Stopping indexing. Waiting for indexers to finish their work (timeout after "
+          <> Text.pack (show secondsBeforeTimeout)
+          <> "s) ..."
+      void $
+        timeout (secondsBeforeTimeout * 1_000_000) $
+          waitQSemN (coordinator ^. barrier) (coordinator ^. indexerCount)
+      logInfo trace "Done!"
```

Comment:
Reapply the same name chaes as in the experimental runner, if we still care about the quality of this part of the codebase.

Reply:
Applied all comments for the experimental runner as well.